kbs/docs/id-key.md

```diff
@@ -0,0 +1,84 @@
+# ID-key Plugin
+
+In addition to fetchable resources upon successful attestation, KBS also offers the ID-key plugin for encrypting and decrypting data with a key provisioned by the KBS. This key can be persisted across multiple reboots of CVMs and help tie decryption resources to a specific KBS instance.
+
+## Data Encryption
+
+As a guest owner would be responsible for encrypting persistent storage, they are the sole authority that is permitted to request resources be encrypted. The KBS is expected to be fully in control by the guest owner with credentials configured.
+
+## Flow
+
+### Provisioning encrypted data
+
+A credentialed guest owner is the only one authorized to request that keys be encrypted by the ID-key plugin. To do this, the guest owner will POST some plaintext data (created locally) to be encrypted. This plaintext data could be a LUKS passphrase, SecureBoot keys, etc. The actual purpose of the data is of no concern to the KBS.
+
+`POST /kbs/v0/id-key/{base64-plaintext-data}`
```

Comment:
would `POST /kbs/v0/id-key` w/ the plaintext in the body make more sense?

Reply:
it's a bit unusual/counter-intuitive for rest-ish APIs to use payload as part of the resource uri in such a request.

Reply:
Ok, I can make the change for this. Perhaps it can just be `/kbs/v0/id-key/register` with the plaintext in the body?